### PR TITLE
Preparation for atomicfu JVM IR plugin application in 1.7.20

### PR DESCRIPTION
**Pull request by mvicsokolova**
- Integration test: excluded Metadata annotation while checking bytecode of class files (`testNoAtomicfuInMppJar`)

Used this solution instead of adding post transformation of metadata by `AtomicfuGradlePlugin`, because the problem is on the side of the compiler plugin: it transforms the backing field of the property, though the property itself remains in the metadata. So, when I'll fix that in the compiler plugin, there will be no atomicfu references left in the metadata.

- Moved `SegmentBasedQueue` class that used inline atomic extension (`findSegmentAndMoveForward`) to the main module.

[Here](https://buildserver.labs.intellij.net/buildConfiguration/Kotlin_BuildPlayground_MilkyWayMargaritaBobova_kotlinxtraincoroutines/216507717?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true) is the sample coroutines build with Integration Test step

### CODE REVIEW

File: integration-testing/src/mavenTest/kotlin/MavenPublicationAtomicfuValidator.kt [hunk 1]
```diff
@@ -33,20 +38,38 @@ class MavenPublicationAtomicfuValidator {
         val foundClasses = mutableListOf<String>()
         for (e in entries()) {
             if (!e.name.endsWith(".class")) continue
+            // check the bytecode of the class for ATOMIC_FU_REF excluding metadata
```
Comment (qwwdfsad) [inline integration-testing/src/mavenTest/kotlin/MavenPublicationAtomicfuValidator.kt:41]: Please drop a comment about when we can remove this workaround and why it is needed in the first place

File: kotlinx-coroutines-core/jvm/test/internal/SegmentListTest.kt [hunk 2]
```diff
@@ -1,41 +1,40 @@
-package kotlinx.coroutines.internal
```
[the remark this answers is not in the document]
Comment (qwwdfsad) [reply kotlinx-coroutines-core/jvm/test/internal/SegmentListTest.kt:1]: I think we can remove it as it's the only way for now.

Whether we need another test or not can be checked by eye-balling the result of Kover: `./gradlew :kotlinx-coroutines-core:koverReport -Pkover.enabled=true`

**Review by qwwdfsad — CHANGES_REQUESTED** [review]

File: kotlinx-coroutines-core/jvm/test/internal/SegmentListTest.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (mvicsokolova) [reply kotlinx-coroutines-core/jvm/test/internal/SegmentListTest.kt:1]: Seems ok, all methods in `ConcurrentLinkedListKt` are covered

**Review by qwwdfsad — APPROVED** [review]
👍

**Pull request merged**